src/Utility/Applications.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * BEdita, API-first content management framework
+ * Copyright 2021 ChannelWeb Srl, Chialab Srl
+ *
+ * This file is part of BEdita: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * See LICENSE.LGPL or <http://gnu.org/licenses/lgpl-3.0.html> for more details.
+ */
+
+namespace App\Utility;
+
+use BEdita\SDK\BEditaClientException;
+use BEdita\WebTools\ApiClientProvider;
+use Cake\Cache\Cache;
+use Cake\Utility\Hash;
+
+/**
+ * Applications class, to get applications data
+ */
+class Applications
+{
+    /**
+     * Get application name by application ID
+     *
+     * @param string $applicationId The application ID
+     * @return string
+     */
+    public static function getName(string $applicationId): string
+    {
+        return (string)Hash::get(self::list(), $applicationId);
+    }
+
+    /**
+     * Get applications, in the form [<id>: <name>].
+     * This uses cache 'applications'.
+     *
+     * @return array
+     */
+    public static function list(): array
+    {
+        try {
+            $applications = Cache::remember(
+                'applications',
```

Comment:
to support multi-project we should `use CacheTrait` and define cache key like:

```php
$this->cacheKey('applications')
```
see [SchemaComponent::getSchema](/bedita/manager/blob/master/src/Controller/Component/SchemaComponent.php#L60)

also a `protected static $applications` array could be used to avoid multiple cache hits